logprep/processor/grokker/processor.py

```diff
@@ -82,27 +82,26 @@ def _apply_rules(self, event: dict, rule: GrokkerRule):
             except TimeoutError as error:
                 self._handle_missing_fields(event, rule, rule.actions.keys(), source_values)
                 raise ProcessingError(
-                    self,
                     f"Grok pattern timeout for source field: '{dotted_field}' in rule '{rule}', "
                     f"the grok pattern might be too complex.",
+                    rule,
                 ) from error
             if result is None or result == {}:
                 continue
             matches.append(True)
-            for dotted_field, value in result.items():
-                if value is None:
-                    continue
-                success = add_field_to(
-                    event, dotted_field, value, rule.extend_target_list, rule.overwrite_target
-                )
-                if not success:
-                    conflicting_fields.append(dotted_field)
+            filtered_items = {k: v for k, v in result.items() if v is not None}
+            targets, contents = zip(*filtered_items.items())
```

Comment:
consider to refactor this together with the code in the `geoip_enricher` with focus on readability.